fhirstore/fhirstore.py

```diff
@@ -91,11 +89,24 @@ def bootstrap(self, resource: Optional[str] = None, show_progress: Optional[bool
                 [
                     ("identifier.system", ASCENDING),
                     ("identifier.value", ASCENDING),
+                ],
+                unique=True,
+                partialFilterExpression={
+                    "identifier.system": {"$exists": True},
+                    "identifier.value": {"$exists": True},
+                },
+            )
+            self.db[resource_type].create_index(
+                [
                     ("identifier.type.coding.system", ASCENDING),
                     ("identifier.type.coding.code", ASCENDING),
+                    ("identifier.value", ASCENDING),
```

Comment:
Needed?

Reply:
Yes, it is! We misunderstood how identifiers with type are built but they always have a value and the "referential" can be indicated whether with system or with both type.coding.system and type.coding.code